precompiles/gov/types.go

```diff
@@ -569,3 +569,77 @@ func (po *ProposalsOutput) FromResponse(res *govv1.QueryProposalsResponse) *Prop
+type ParamsOutput struct {
+	VotingPeriod               int64      `abi:"votingPeriod"`
+	MinDeposit                 []cmn.Coin `abi:"minDeposit"`
+	MaxDepositPeriod           int64      `abi:"maxDepositPeriod"`
+	Quorum                     string     `abi:"quorum"`
+	Threshold                  string     `abi:"threshold"`
+	VetoThreshold              string     `abi:"vetoThreshold"`
+	MinInitialDepositRatio     string     `abi:"minInitialDepositRatio"`
+	ProposalCancelRatio        string     `abi:"proposalCancelRatio"`
+	ProposalCancelDest         string     `abi:"proposalCancelDest"`
+	ExpeditedVotingPeriod      int64      `abi:"expeditedVotingPeriod"`
+	ExpeditedThreshold         string     `abi:"expeditedThreshold"`
+	ExpeditedMinDeposit        []cmn.Coin `abi:"expeditedMinDeposit"`
+	BurnVoteQuorum             bool       `abi:"burnVoteQuorum"`
+	BurnProposalDepositPrevote bool       `abi:"burnProposalDepositPrevote"`
+	BurnVoteVeto               bool       `abi:"burnVoteVeto"`
+	MinDepositRatio            string     `abi:"minDepositRatio"`
+}
+
+// FromResponse populates the ParamsOutput from a query response
+func (o *ParamsOutput) FromResponse(res *v1.QueryParamsResponse) *ParamsOutput {
+	o.VotingPeriod = res.Params.VotingPeriod.Nanoseconds()
+	o.MinDeposit = cmn.NewCoinsResponse(res.Params.MinDeposit)
+	o.MaxDepositPeriod = res.Params.MaxDepositPeriod.Nanoseconds()
+	o.Quorum = res.Params.Quorum
+	o.Threshold = res.Params.Threshold
+	o.VetoThreshold = res.Params.VetoThreshold
+	o.MinInitialDepositRatio = res.Params.MinInitialDepositRatio
+	o.ProposalCancelRatio = res.Params.ProposalCancelRatio
+	o.ProposalCancelDest = res.Params.ProposalCancelDest
+	o.ExpeditedVotingPeriod = res.Params.ExpeditedVotingPeriod.Nanoseconds()
+	o.ExpeditedThreshold = res.Params.ExpeditedThreshold
+	o.ExpeditedMinDeposit = cmn.NewCoinsResponse(res.Params.ExpeditedMinDeposit)
+	o.BurnVoteQuorum = res.Params.BurnVoteQuorum
+	o.BurnProposalDepositPrevote = res.Params.BurnProposalDepositPrevote
+	o.BurnVoteVeto = res.Params.BurnVoteVeto
+	o.MinDepositRatio = res.Params.MinDepositRatio
+	return o
+}
+
+// ParseParamsArgs parses the arguments for the params query
+func ParseParamsArgs(args []interface{}) (*v1.QueryParamsRequest, error) {
+	if len(args) != 1 {
+		return nil, fmt.Errorf(cmn.ErrInvalidNumberOfArgs, 1, len(args))
+	}
+
+	paramsType, ok := args[0].(string)
+	if !ok {
+		return nil, fmt.Errorf(cmn.ErrInvalidType, "paramsType", "string", args[0])
+	}
+
+	return &v1.QueryParamsRequest{
+		ParamsType: paramsType,
+	}, nil
+}
+
+// NewMsgUpdateParams creates a new MsgUpdateParams instance
+func NewMsgUpdateParams(authority string, args []interface{}) (*v1.MsgUpdateParams, error) {
+	if len(args) != 1 {
+		return nil, fmt.Errorf(cmn.ErrInvalidNumberOfArgs, 1, len(args))
+	}
+
+	params, ok := args[0].(v1.Params)
+	if !ok {
+		return nil, fmt.Errorf(cmn.ErrInvalidType, "params", "Params", args[0])
+	}
```

Comment:
remove

Reply:
this whole logic seems weird, we enforce passing one variable but then again force it to be empty. That seems like bad design